lib/pkcs7.js

```diff
@@ -342,16 +342,20 @@ p7.createSignedData = function() {
         if('content' in msg) {
           var content;
           if(msg.content instanceof forge.util.ByteBuffer) {
-            content = msg.content.bytes();
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, false,
+              msg.content.bytes());
           } else if(typeof msg.content === 'string') {
-            content = forge.util.encodeUtf8(msg.content);
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, false,
+              forge.util.encodeUtf8(msg.content));
+          } else if (typeof msg.content === 'object') {
+            content = asn1.create(asn1.Class.UNIVERSAL, asn1.Type.OCTETSTRING, true,
+              [msg.content]);
```

Comment:
I'm also not yet sure how my above comment meshes with parsing PKCS#7 messages, so I'll have to look back at that.